evalml/tests/automl_tests/test_automl.py

```diff
@@ -2155,10 +2246,22 @@ def clone(self):
         y_train=y,
         problem_type="binary",
         max_iterations=2,
-        allowed_pipelines=[DummyPipeline({})],
         objective="log loss binary",
         additional_objectives=["f1"],
     )
+    automl._automl_algorithm = IterativeAlgorithm(
```

Comment:
Why is this needed?

Reply:
Because the `DummyPipeline` score is being mocked for the purpose of this test (which is something ComponentGraph doesn't have), I have to specify that pipeline being passed into the `IterativeAlgorithm` since we can't pass it through `AutoMLSearch` anymore. This is the same reason it was added for the other tests as well